### ISSUE TITLE
What to do about stale data?

### ISSUE THREAD

**Issue by skrzepto** [skrzepto/IPRO-VAULT-SERVER#7]
If an update has not come in after 30 minutes to the last update (frequency *2 time aka 15 min *2 = 30 min) Do we want to push the node to the fault table and state reason as `stale data`

**Comment by sdimitro** [comment]
Yeah that sounds reasonable. We can easily do that by changing our shared memory table implementation of the table to a table manipulated through a pipeline.

http goroutines: we have those already.
supervisor goroutine: the only goroutine that manipulates the table (we no longer need Locks)
timeout goroutine: has an entry per sensor that contains the most recent timestamp, and notifies supervisor of timeouts.

Pipeline:
http request goroutine --sensor data--> supervisor goroutine <--time notifications--  --timestamp data --> timeout goroutine.
